scanner/src/test/java/edu/ucr/cs/riple/scanner/MethodInfoTest.java

```diff
@@ -90,6 +92,61 @@ public void BasicTest() {
                 "0",
                 "[]",
                 "false",
+                "public",
+                "[]",
+                "edu/ucr/A.java"))
+        .doTest();
+  }
+
+  @Test
+  public void visibilityTest() {
```

Comment:
I can think of another three worthwhile cases to test:

1. Protected methods
2. Methods with modifiers unrelated to visibility (such as `default`, `abstract`, `static`, etc) in addition to visibility modifiers. No need to test all combinations, but it's worth having an example that has more than a single modifier on a method signature.
3. Methods of an interface, rather than a class. Does this approach figure out that "public" is [implicit](https://docs.oracle.com/javase/specs/jls/se8/html/jls-9.html#jls-9.4) for interface methods? I think it might. I think that information may already be explicit at the compiler level, but we should check.

My personal instinct would be to add (1) to the test case below, and make new separate test cases for (2) and (3).

Reply:
I added the method return type as well. Just a flag showing the return type is non-primitive. In the upstream analysis we should not waste builds with methods which are return `void/primitive`. Updated this PR description as well.

edccb99ef8288dfcb90cbc155a727986340f7998